src/index.js

```diff
@@ -7,26 +7,48 @@ var path = require('path')
 var request = require('request')
 var tarFS = require('tar-fs')
 var unzip = require('unzip')
-var version = require('./../package.json').version
+var pkg  = require('./../package.json')
 
-module.exports = function (callback) {
+module.exports = function (version, platform, callback) {
+  if (typeof version === 'function') {
+    callback = version
+    version = 'v' + pkg.version
+  }
+
+  if (typeof platform === 'function') {
+    callback = platform
+    platform = goenv.GOOS
+  }
+
+  // Default values
+  version  = version  || process.env.TARGET_VERSION || 'v' + pkg.version
+  platform = platform || process.env.TARGET_OS      ||goenv.GOOS
   callback = callback || noop
 
   // make sure we can do this.
   if (!checkPlatform.isSupportedArchitecture(goenv.GOARCH)) {
     throw new Error('no binary available for arch: ' + goenv.GOARCH)
   }
 
-  const isWindows = checkPlatform.isWindows(goenv.GOOS)
+  // Set the platform either from arguments or goenv OS
+  const isWindows = checkPlatform.isWindows(platform)
 
   // hacky hack hack to work around unpublishability
-  version = version.replace(/-[0-9]+/, '')
+  // version = version.replace(/-[0-9]+/, '')
+  // What is this^^?? @haadcode
```

Comment:
Not sure we can drop this, because the version in `package.json` corresponds to go version, but we sometimes have multiple versions of this package per go-ipfs version, so this fixes that, i.e.

| go-ipfs | package.json |
| 0.4.4    | 0.4.4    |
| 0.4.4    | 0.4.4-1 | 
| 0.4.4    | 0.4.4-2 |
| 0.4.4    | 0.4.4-3 |